### ISSUE TITLE
Improve/fix current API spec to make it parseable by Postman/janephp tools

### ISSUE THREAD

**Issue by vladdancer** [Picturepark/Picturepark.SDK.Samples#194]
Hi PP team!

I assume that this repository is better suited for this issue.

I'm going integrate PP with Drupal for our client.
I found that current Swagger APi is too heavy and looks broken for me.
I've tried to import current spec from here https://api-cp-ch01.picturepark.com/docs/rest/PictureparkSwagger.json into Postman and got error in the middle of the process.

The API spec version: (NSwag v13.6.2.0 (NSwag v13.13.2.0 (NJsonSchema v10.5.2.0 (Newtonsoft.Json v12.0.0.0)))

I've also tried to generate automatically PHP classes for request/responses using [janephp]( https://github.com/janephp/janephp) tool but getting next error https://github.com/janephp/janephp/issues/445
So I assume that current spec is a bit broken because two tools can't parse it.
But you folks somehow work with it in the team that's why I wonder how are you dealing with this spec since it is soooo huuuuge and broken?

BTW: why are you rendring whole huge api spec here https://api-cp-ch01.picturepark.com/docs/Rest/index.html#section/Getting-started 
Instead of current way, you could provide list of call and then call details/docs on click by ajax + integrating algolia search on the page or https://docs.meilisearch.com is good option too. 

Thank you in advance!

**Comment by vladdancer** [comment]
Hi @neotek.

I finally managed what problem is, at least with current PP spec "NSwag v13.14.8.0 (NJsonSchema v10.5.2.0 (Newtonsoft.Json v13.0.0.0))"  and jane-php/jane-php:v7.4.0, jane-php/open-api-3:v7.4.0.

TL;DR
`"description": null` in specific place gives an error:
```
Jane\Component\JsonSchemaRuntime\Reference::__construct(): Argument #2 ($origin) must be of type string, null given, called in vendor/jane-php/jane-php/src/Component/OpenApi3/JsonSchema/Normalizer/ReferenceNormalizer.php on line 35 and defined in vendor/jane-php/jane-php/src/Component/JsonSchemaRuntime/Reference.php:30
```

So the next json is "wrong" in terms of Jane.
```
{
  "400": {
    "description": null,
    "content": {
      "application/json": {
        "schema": {
          "$ref": "#/components/schemas/PictureparkBusinessException"
        }
      }
    }
  }
}
```

So there are at least 7 occurences with empty description where content is application/json and schema has $ref
- /v1/Contents/{id}:delete
- /v1/JsonSchemas/{id}:get
- /v1/ListItems/{id}:delete
- /v1/Notifications/emailNotifications/types:get
- /v1/SchemaTransfers/import:post
- /v1/Shares:post
- /v1/Statistics/contents/{contentId}:get

Currently it is not a problem for me. I'm just replacing all bad occurences with sed:
`sed -i 's/"description": null/"description": "null"/g' PictureparkSwagger.json`

**Comment by neotek** [comment]
Hi @vladdancer 

Thanks for your feedback.
I created an issue for this so we can fix it in a later release.

Best regards

**Comment by vladdancer** [comment]
Also got one more issue: circular reference.
```
    "/v1/Shares": {
      "post": {
        "operationId": "Share_Create",
        "requestBody": {
          "x-name": "request",
          "description": "Polymorphic create contract. Use either ShareBasicCreateRequest or ShareEmbedCreateRequest",
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/ShareBaseCreateRequest"
              }
            }
          },
          "required": true,
          "x-position": 1
        },
```

```
      "ShareBaseCreateRequest": {
        "type": "object",
        "discriminator": {
          "propertyName": "kind",
          "mapping": {
            "ShareBasicCreateRequest": "#/components/schemas/ShareBasicCreateRequest",
            "ShareEmbedCreateRequest": "#/components/schemas/ShareEmbedCreateRequest"
          }
        },
```

and `components.schemas.ShareEmbedCreateRequest`
```
"ShareEmbedCreateRequest": {
        "allOf": [
          {
            "$ref": "#/components/schemas/ShareBaseCreateRequest"
          },
          {
            "type": "object",
            "description": "Create request for embed share",
            "additionalProperties": false
          }
        ]
      },
```
causes infinite loop (Segmentation fault) in a code generated by janephp: \Picturepark\Api\Generated\Normalizer\ShareBaseCreateRequestNormalizer
```
    public function normalize($object, $format = null, array $context = array())
    {
        $data = array();
        if (null !== $object->getKind() and 'ShareBasicCreateRequest' === $object->getKind()) {
            return $this->normalizer->normalize($object, $format, $context);
        }
        if (null !== $object->getKind() and 'ShareEmbedCreateRequest' === $object->getKind()) {
            return $this->normalizer->normalize($object, $format, $context);
        }
       ...
  }
```
